hera_cal/abscal.py

```diff
@@ -0,0 +1,484 @@
+                 Can be generated from a pyuvdata.UVData instance via
+                 ----
+                 #!/usr/bin/env python
+                 uvd = pyuvdata.UVData()
+                 uvd.read_miriad(<filename>)
+                 antenna_pos, ants = uvd.get_ENU_antpos()
+                 antpos = dict(zip(ants, antenna_pos))
+                 ----
+                 This is needed only for Tip Tilt phase calibration.
+
+        freqs : ndarray of frequency array, type=ndarray, dtype=float
+            1d array containing visibility frequencies in Hz. Needed to write gain solutions
+            out to calfits.
+    
+        times : ndarray of time array, type=ndarray, dtype=float
+            1d array containing visibility times in Julian Date. Needed to write out to calfits.
+
+        pols : ndarray of polarization array, type=ndarray, dtype=int
+            array containing polarization integers in pyuvdata.UVData.polarization_array 
+            format. Needed to write out to calfits.
+        """
+        # get shared keys
+        self.keys = sorted(set(model.keys()) & set(data.keys()))
+
+        # append attributes
+        self.model = model
+        self.data = data
+
+        # setup frequencies
+        self.freqs = freqs
+        if self.freqs is None:
+            self.Nfreqs = None
+        else:
+            self.Nfreqs = len(self.freqs)
+
+        # setup polarization
+        self.str2pol = {"xx": -5, "yy": -6, "xy": -7, "yx": -8}
+        if pols is not None:
+            if type(pols) is list:
+                if type(pols[0]) is str:
+                    pols = map(lambda x: self.str2pol[x], pols)
+                elif type(pols[0]) is int:
+                    pols = pols
+            elif type(pols) is str:
+                pols = [self.str2pol[pols]]
+            elif type(pols) is int:
+                pols = [pols]
+            self.pols = pols
+            self.Npols = len(pols)
+        else:
+            self.pols = None
+            self.Npols = None
+
+        # setup weights
+        if wgts is None:
+            wgts = odict()
+            for i, k in enumerate(self.keys):
+                wgts[k] = np.ones_like(data[k], dtype=np.float)
+        self.wgts = wgts
+
+        # setup times
+        self.times = times
+        if times is None:
+            self.Ntimes = None
+        else:
+            self.Ntimes = len(self.times)
+
+        # setup ants
+        self.ants = np.unique(np.concatenate(map(lambda k: k[:2], self.keys)))
+        self.Nants = len(self.ants)
+
+        # setup baselines and antenna positions
+        self.antpos = antpos
+        self.bls = None
+        if self.antpos is not None:
+            self.bls = odict([(x, self.antpos[x[1]] - self.antpos[x[0]]) for x in self.keys])
+            self.antpos = np.array(map(lambda x: self.antpos[x], self.ants))
+            self.antpos -= np.median(self.antpos, axis=0)
+
+    def amp_logcal(self, separate_pol=False, verbose=True):
```

Comment:
Separate_pol should default to True, since that's what works with 2pol remove degen, our default operating mode.

Reply:
So the reason why I defaulted it to False, is because if you are feeding only a single polarization (which most of our data conforms to) then it is the same thing, and because `separate_pol` requires the data to be copied over it will slow things down when working with large files (HERA-47 AbsCal is already not very fast).